### PR TITLE
Be consistent in testing outputs from railties test and use /bin/rails

### PR DESCRIPTION
**Pull request by vipulnsward**
Be consistent in testing outputs from railties test and use /bin/rails everywhere(the default behaviour now) instead of mix of /bin/rake and /bin/rails everywhere

### CODE REVIEW

File: railties/test/application/test_runner_test.rb [hunk 1]
```diff
@@ -446,7 +446,7 @@ def test_pass_TEST_env_on_rake_test
       create_test_file :models, 'account'
       create_test_file :models, 'post', pass: false
 
-      output =  Dir.chdir(app_path) { `bin/rake test TEST=test/models/post_test.rb` }
+      output =  Dir.chdir(app_path) { `bin/rails test TEST=test/models/post_test.rb` }
```
Comment (vipulnsward) [inline railties/test/application/test_runner_test.rb]: @senny is this supposed to be supported only on rake, or this should be fixed on `/bin/rails` as well?
https://travis-ci.org/rails/rails/jobs/104328919#L927

Comment (vipulnsward) [inline railties/test/application/test_runner_test.rb]: Original fix was https://github.com/vipulnsward/rails/commit/64448c29deb2f7239ac7e21035c05b8793ee1f47 , works for rake only.

Comment (kaspth) [inline railties/test/application/test_runner_test.rb]: This test is meant specifically to test `rake test` (see the name) and shouldn't be changed.

Comment (vipulnsward) [inline railties/test/application/test_runner_test.rb]: @kaspth true, thats why the question. Is this supposed to be supported only on rake? This seems to be the only option this way.

**Comment by kaspth** [comment]
Builds on #23172, @vipulnsward can you add @ryohashimoto to your commit message so he'll get credit as well? Like this: http://contributors.rubyonrails.org/faq#credit-several-people

File: railties/test/application/test_runner_test.rb (hunk 1 above)
Comment (kaspth) [inline railties/test/application/test_runner_test.rb]: `TEST` is only for backwardscompatibility with `rake test` as `bin/rails test` already supports running tests from a single file more cleanly.

**Comment by vipulnsward** [comment]
Updated. Hope `[Ryo Hashimoto & Vipul A M]` works.

**Comment by kaspth** [comment]
@vipulnsward sweet, those are your names on the contributors site, so I think it will. Thanks!

Now we just need the stray rails -> rake fixed and we're good to go 👍

File: railties/test/application/test_runner_test.rb [hunk 2]
```diff
@@ -445,7 +445,8 @@ def test_raise_error_when_specified_file_does_not_exist
     def test_pass_TEST_env_on_rake_test
       create_test_file :models, 'account'
       create_test_file :models, 'post', pass: false
-
+      # This specifically verifies TEST for backwards compatibility with rake test
+      # as bin/rails test already supports running tests from a single file more cleanly.
```
Comment (vipulnsward) [inline railties/test/application/test_runner_test.rb]: Ok, specified why this is a special test.

Comment (kaspth) [inline railties/test/application/test_runner_test.rb]: Personally don't think this adds much. The fact that it's run with `bin/rake` should be enough of a clue. But I'll let the build run and if it's green, I'll merge 👍

**Comment by ryohashimoto** [comment]
Nice work. Thanks for adding me in the commit message.

**Pull request merged**